components/autofill/sql/create_shared_schema.sql

```diff
@@ -55,36 +55,40 @@ CREATE TABLE IF NOT EXISTS addresses_tombstones (
 -- whether the `cc_number` and/or other details should be encrypted or stored as plain text. Currently, we are storing
 -- them as plain text.
 CREATE TABLE IF NOT EXISTS credit_cards_data (
-    guid                TEXT NOT NULL PRIMARY KEY,
-    cc_name             TEXT NOT NULL, -- full name
-    cc_number           TEXT NOT NULL, -- TODO: consider storing this field as a hash
-    cc_exp_month        INTEGER,
-    cc_exp_year         INTEGER,
-    cc_type             TEXT NOT NULL,
-    -- cc_exp              TEXT NOT NULL, -- text format of the expiration date e.g. "[cc_exp_year]-[cc_exp_month]"
+    guid                    TEXT NOT NULL PRIMARY KEY,
+    billing_address_guid    TEXT NOT NULL,
+    cc_name                 TEXT NOT NULL, -- full name
+    cc_number               TEXT NOT NULL, -- TODO: consider storing this field as a hash
+    cc_exp_month            INTEGER,
+    cc_exp_year             INTEGER,
+    cc_type                 TEXT NOT NULL,
 
-    time_created        INTEGER NOT NULL,
-    time_last_used      INTEGER,
-    time_last_modified  INTEGER NOT NULL,
-    times_used          INTEGER NOT NULL DEFAULT 0,
+    time_created            INTEGER NOT NULL,
+    time_last_used          INTEGER,
+    time_last_modified      INTEGER NOT NULL,
+    times_used              INTEGER NOT NULL DEFAULT 0,
 
     /* Same "sync change counter" strategy used by other components. */
-    sync_change_counter INTEGER NOT NULL DEFAULT 1
+    sync_change_counter INTEGER NOT NULL DEFAULT 1,
+
+    FOREIGN KEY(billing_address_guid) REFERENCES addresses_data(guid) ON DELETE SET DEFAULT
```

Comment:
@mhammond  FYI, I chose `SET DEFAULT` as the `ON DELETE` action based on our conversation earlier about the one-to-many relationship between addresses and credit cards as well as [this test assertion](https://searchfox.org/mozilla-central/source/browser/extensions/formautofill/test/unit/test_creditCardRecords.js#352) in desktop. But it might make more sense to cascade.

Reply:
How does this work given `billing_address_guid` is `NOT NULL`?